test/e2e/flask/solana/common-solana.ts

```diff
@@ -0,0 +1,72 @@
+import { Mockttp } from 'mockttp';
+import { withFixtures, unlockWallet } from '../../helpers';
+import { Driver } from '../../webdriver/driver';
+import HeaderNavbar from '../../page-objects/pages/header-navbar';
+import AccountListPage from '../../page-objects/pages/account-list-page';
+import FixtureBuilder from '../../fixture-builder';
+import { ACCOUNT_TYPE } from '../../page-objects/common';
+
+const SOLANA_URL_REGEX = /^https:\/\/.*\.solana.*/u;
+
+export enum SendFlowPlaceHolders {
+  AMOUNT = 'Enter amount to send',
+  RECIPIENT = 'Enter receiving address',
+  LOADING = 'Preparing transaction',
+}
+
+export async function mockSolanaBalanceQuote(mockServer: Mockttp) {
+  return await mockServer
+    .forPost(SOLANA_URL_REGEX)
+    .withJsonBodyIncluding({
+      method: 'getBalance',
+    })
+    .thenCallback(() => {
+      return {
+        statusCode: 200,
+        json: {
+          result: {
+            context: {
+              apiVersion: '2.0.15',
+              slot: 305352614,
+            },
+            value: 0,
+          },
+        },
+      };
+    });
+}
+
+export async function withSolanaAccountSnap(
+  {
+    title,
+    solanaSupportEnabled,
+  }: { title?: string; solanaSupportEnabled?: boolean },
+  test: (driver: Driver, mockServer: Mockttp) => Promise<void>,
+) {
+  console.log('Starting withSolanaAccountSnap');
+  await withFixtures(
+    {
+      fixtures: new FixtureBuilder()
+        .withPreferencesControllerAndFeatureFlag({
+          solanaSupportEnabled: solanaSupportEnabled ?? true,
+        })
+        .build(),
+      title,
+      dapp: true,
+      testSpecificMock: async (mockServer: Mockttp) => {
+        console.log('Setting up test-specific mocks');
+        return [await mockSolanaBalanceQuote(mockServer)];
+      },
+    },
+    async ({ driver, mockServer }: { driver: Driver; mockServer: Mockttp }) => {
+      await unlockWallet(driver);
+      const headerComponen = new HeaderNavbar(driver);
+      console.log('Opening account menu');
+      await headerComponen.openAccountMenu();
+      const accountListPage = new AccountListPage(driver);
+      await accountListPage.addAccount(ACCOUNT_TYPE.SOLANA, 'Solana 1');
```

Comment:
I believe that you don't need to specify `Solana 1`, since the Snap should take care of this automatically but maybe the naming is different for Solana?

We can keep it if you prefer! :)